src/sync/sys/windows/net.rs

```diff
@@ -0,0 +1,303 @@
+            instance_number: 0, //todo for client scenarios
+        }
+    }
+}
+
+impl PipeConnection {
+    pub(crate) fn id(&self) -> i32 {
+        self.instance_number
+    }
+
+    pub fn read(&self, buf: &mut [u8]) -> Result<usize> {
+        trace!("reading from  pipe: {}", self.instance_number);
+
+        let mut events = Events::with_capacity(1024);
+        loop {
+            // grabbing the lock on the poller here isn't ideal but the named pipe needs mutable access to read
+            // This is ok though as read is currently blocking other threads in the server impl (only one read at a time)
+            // It is also blocking until read event comes through.  This is preferable as it will not cause any cpu cycles 
+            self.poller.lock().unwrap().poll(&mut events, None).unwrap();
+            match self.named_pipe.lock().unwrap().read(buf) {
+                Ok(0) => {
+                    return Err(crate::Error::LocalClosed);
+                }
+                Ok(x) => {
+                    //print!("read: {:?}", std::str::from_utf8(&buf));
+                    return Ok(x);
+                }
+                Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                    continue;
+                }
+                Err(e) if e.raw_os_error().is_some() => {
+                    return Err(crate::Error::Windows(e.raw_os_error().unwrap()))
+                }
+                Err(e) => {
+                    trace!("Error writing to pipe: {}", e);
+                    return Err(crate::Error::Others(e.to_string()));
+                }
+            }
+        }
+    }
+
+    pub fn write(&self, buf: &[u8]) -> Result<usize> {
+        trace!("Writing to  pipe: {}", self.instance_number);
+        loop {
+            // grabbing the lock write to read isn't ideal
+            // the named pipe needs mutable access to read
+            match self.named_pipe.lock().unwrap().write(buf) {
+                Ok(x) => return Ok(x),
+                Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                    continue;
+                }
+                Err(e) if e.raw_os_error() == Some(ERROR_NO_DATA as i32) => {
+                    return Err(crate::Error::Windows(e.raw_os_error().unwrap()))
+                }
+                Err(e) if e.raw_os_error().is_some() => {
+                    return Err(crate::Error::Windows(e.raw_os_error().unwrap()))
+                }
+                Err(e) => {
+                    trace!("Error writing to pipe: {}", e);
+                    return Err(crate::Error::Others(e.to_string()));
+                }
+            }
+        }
+    }
+
+    pub fn close(&self) -> Result<()> {
+        let h = self.named_pipe.lock().unwrap().as_raw_handle();
+        let result = unsafe { CloseHandle(h as isize) };
+        
+
+        let os_err = unsafe {  GetLastError() as i32 };
+        match result {
+            0 => Err(crate::Error::Windows(os_err)),
```

Comment:
See https://github.com/containerd/ttrpc-rust/pull/172/files#r1114060631, but can we do the below? 
```suggestion
            0 => Err(Error::last_os_error()),
```

Reply:
most of the package uses the `crate::Error` (moved this to an import too) through out, so still need that but did switch to the `last_os_error`